doc/guides/device_mgmt/index.rst

```diff
@@ -39,14 +39,466 @@ functionality over BLE and serial.
+available sub-commands are::
+
+  mcumgr <connection-options> stat list
+  mcumgr <connection-options> stat <section-name>
+
+Statistics are organized in sections (also called groups), and each section can
+be individually queried. Defining new statistics sections is done by using macros
+available under ``<stats/stats.h>``. Each section consists of multiple variables
+(or counters), all with the same size (16, 32 or 64 bits).
+
+To create a new section ``my_stats``::
+
+  STATS_SECT_START(my_stats)
+    STATS_SECT_ENTRY(my_stat_counter1)
+    STATS_SECT_ENTRY(my_stat_counter2)
+    STATS_SECT_ENTRY(my_stat_counter3)
+  STATS_SECT_END;
+
+  STATS_SECT_DECL(my_stats) my_stats;
+
+Each entry can be declared with ``STATS_SECT_ENTRY`` (or the equivalent
+``STATS_SECT_ENTRY32``), ``STATS_SECT_ENTRY16`` or ``STATS_SECT_ENTRY64``.
+All statistics in a section must be declared with the same size.
+
+The statistics counters can either have names or not, depending on the setting
+of the :option:`CONFIG_STATS_NAMES` option. Using names requires an extra
+declaration step::
+
+  STATS_NAME_START(my_stats)
+    STATS_NAME(my_stats, my_stat_counter1)
+    STATS_NAME(my_stats, my_stat_counter2)
+    STATS_NAME(my_stats, my_stat_counter3)
+  STATS_NAME_END(my_stats);
+
+.. tip::
+
+   Disabling :option:`CONFIG_STATS_NAMES` will free resources. When this option
+   is disabled the ``STATS_NAME*`` macros output nothing, so adding them in the
+   code does not increase the binary size.
+
+.. tip::
+
+   :option:`CONFIG_STAT_MGMT_MAX_NAME_LEN` sets the maximum length of a section
+   name that can can be accepted as parameter for showing the section data, and
+   might require tweaking for long section names.
+
+The final steps to use a statistics section is to initialize and register it::
+
+  rc = STATS_INIT_AND_REG(my_stats, STATS_SIZE_32, "my_stats");
+  assert (rc == 0);
+
+In the running code a statistics counter can be incremented by 1 using
+``STATS_INC``, by N using ``STATS_INCN`` or reset with ``STATS_CLEAR``.
+
+Let's suppose we want to increment those counters by ``1``, ``2`` and ``3``
+every second. To get a list of stats::
+
+  $ mcumgr --conn acm0 stat list
+  stat groups:
+    my_stats
+
+To get the current value of the counters in ``my_stats``::
+
+  $ mcumgr --conn acm0 stat my_stats
+  stat group: my_stats
+        13 my_stat_counter1
+        26 my_stat_counter2
+        39 my_stat_counter3
+
+  $ mcumgr --conn acm0 stat my_stats
+  stat group: my_stats
+        16 my_stat_counter1
+        32 my_stat_counter2
+        48 my_stat_counter3
+
+When :option:`CONFIG_STATS_NAMES` is disabled the output will look like this::
+
+  $ mcumgr --conn acm0 stat my_stats
+  stat group: my_stats
+         8 s0
+        16 s1
+        24 s2
+
+.. _fs_mgmt:
+
+Filesystem Management
+*********************
+
+The filesystem module is disabled by default because of security concerns,
+because of a lack of access control every file in the FS will be accessible,
```

Comment:
```suggestion
The filesystem module is disabled by default due to security concerns:
because of a lack of access control every file in the FS will be accessible,
```

Reply:
Done.